gpytorch/lazy/lazy_variable.py

```diff
@@ -1,9 +1,87 @@
 class LazyVariable(object):
     def evaluate(self):
+        """
+        Explicitly evaluates the matrix this LazyVariable represents. This
+        function should return a Variable explicitly wrapping a Tensor storing
+        an exact representation of this LazyVariable.
+        """
         raise NotImplementedError
 
     def add_diag(self, diag):
+        """
+        Adds a diagonal component to this lazy variable in a potentially lazy fashion.
+        That is, if this lazy variable represents some matrix A, add_diag should return
+        a LazyVariable of the same type that represents A+dI.
+
+        Args:
+            - diag (scalar) - diagonal component to add.
+        Returns:
+            - LazyVariable representing this current LazyVariable plus diag*I.
+        """
         raise NotImplementedError
 
     def gp_marginal_log_likelihood(self, target):
+        """
+        Computes the marginal log likelihood of a Gaussian process whose covariance matrix
+        plus the diagonal noise term (added using add_diag above) is stored as this lazy variable
+
+        Args:
+            - target (vector n) - training label vector to be used in the marginal log likelihood calculation.
+        Returns:
+            - scalar - The GP marginal log likelihood where (K+\sigma^{2}I) is represented by this LazyVariable.
+        """
+        raise NotImplementedError
+
+    def mvn_kl_divergence(self, mean_1, chol_covar_1, mean_2):
```

Comment:
This might not be necessary with the latest push into master.
See the new `mvn_kl_divergence` in `gpytorch.__init__`